### PR TITLE
Fix goroutine leak on logs -f with no output

### PR DESCRIPTION
**Pull request by cpuguy83**
Also noticed while making this that if you use the API with follow and only 1 of stdout or stderr (not both), the log routine would also hang.

### CODE REVIEW

File: daemon/logs.go [hunk 1]
```diff
@@ -131,27 +133,46 @@ func (daemon *Daemon) ContainerLogs(name string, config *ContainerLogsConfig) er
 			stdoutPipe = container.StdoutLogPipe()
 			go func() {
 				logrus.Debug("logs: stdout stream begin")
-				chErr <- jsonlog.WriteLog(stdoutPipe, outStream, format, config.Since)
+				chErrStdout <- jsonlog.WriteLog(stdoutPipe, outStream, format, config.Since)
 				logrus.Debug("logs: stdout stream end")
 			}()
+		} else {
+			close(chErrStdout)
```
Comment (LK4D4) [inline daemon/logs.go]: Doesn't this channel return immediately in select then?

Comment (cpuguy83) [inline daemon/logs.go]: ah yeah, you're right... let me fix.

Comment (cpuguy83) [inline daemon/logs.go]: updated

**Comment by LK4D4** [comment]
LGTM

**Comment by estesp** [comment]
Looks like a as good a test as you can construct for the scenario, but should there be a comment that if the test suite is ever parallelized that this check won't work anymore; or at least won't be doing what we think it does today?

**Comment by LK4D4** [comment]
I think all tests won't work under parallelization :)

**Comment by estesp** [comment]
ha, very true I guess.. it was in my head because something I was reading about Go 1.5 and parallelized building.. but, yeah, we have bigger worries than just this test for that scenario.

**LGTM**

**Pull request merged**